### ISSUE TITLE
Can not filter by “Duration (μs)> =”  in the UI

### ISSUE THREAD

**Issue by longinhit** [openzipkin/zipkin#1349]

**Comment by codefromthecrypt** [comment]
the duration query depends a lot on how your apps are instrumented and
which storage is in use. For example, we no longer support duration query
on the "cassandra" storage type.

**Comment by michaelsembwever** [comment]
@longinhit it's supported in the cassandra3 storage, if you're able to upgrade.

**Comment by dangets** [comment]
Is this the same issue with the Elasticsearch backend?  Results that shouldn't be are getting filtered out.

![zipkin 1](https://cloud.githubusercontent.com/assets/753632/20492871/b660dd40-afd3-11e6-8e0e-ef9dc3bc1daa.png)
![zipkin 2](https://cloud.githubusercontent.com/assets/753632/20492875/b9dc6e8a-afd3-11e6-988b-7faf43e6beb1.png)

**Comment by codefromthecrypt** [comment]
what instrumentation are you using? are the traces you are sending
reporting span.duration? You can check by using the api as
/api/v1/trace/trace-id-in-hex?raw

**Comment by codefromthecrypt** [comment]
fyi here's a status report on libraries that properly address
timestamp and duration
https://github.com/openzipkin/openzipkin.github.io/issues/49

**Comment by dangets** [comment]
We are using brave v3.15.1.  We were previously using v3.9.1 until a few weeks ago and that is around when we started seeing the problem (upgraded brave and zipkin-server jar).

We do use a custom jmeter setup to run automated tests and we generate a traceId before sending the initial request so that we can log traceIds alongside request logs.  Using the raw endpoint I am seeing sr and ss on the top-level span but it looks like the duration field is missing from that span.

Again, this was working previously - is this something that has changed during the deprecation of the `SpanCollector` classes?  I can try switching to the `Reporter` classes to see if that fixes it.

**Comment by dangets** [comment]
I verified that when I hit an endpoint using curl and specifying a traceId in the headers the duration does _not_ show up.  Without the headers it does.

And we found this comment that says a timestamp/duration won't be kept if a traceId is sent - perhaps the cause? https://github.com/openzipkin/brave/blob/181a2f438e5142510cb72eefc723839bc525fe4d/brave-core/src/main/java/com/github/kristofa/brave/ServerRequestInterceptor.java#L50

Does a backfilling as mentioned in the comment ever happen?
